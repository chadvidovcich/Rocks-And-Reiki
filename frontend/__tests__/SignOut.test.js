// @ts-ignore
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MockedProvider } from '../node_modules/@apollo/react-testing/index';
import SignOut, { SIGNOUT_MUTATION } from '../components/SignOut';
import wait from 'waait';

const mocks = [
  // mutation mock
  {
    request: {
      query: SIGNOUT_MUTATION,
      variables: {},
    },
    result: {
      success: [{ message: 'Success Log Out' }],
    },
  },
];

describe('<SignOut/>', () => {
  it('should render and match snapshot', () => {
    const { container, debug } = render(
      <MockedProvider>
        <SignOut />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });
  it('should call the mutation properly', async () => {
    const { container, debug } = render(
      <MockedProvider mocks={mocks}>
        <SignOut />
      </MockedProvider>
    );
    await wait(400);
    // TODO
    // click log out button
    // await userEvent.click(screen.getByTestId('logOutButton'));
    // debug();
  });
});
