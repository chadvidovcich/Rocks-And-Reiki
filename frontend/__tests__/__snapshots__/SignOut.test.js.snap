// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignOut/> should render and match snapshot 1`] = `
<div>
  <button
    data-testid="logOutButton"
    type="button"
  >
    Sign Out
  </button>
</div>
`;
